sw/battor.c

```diff
@@ -104,6 +104,21 @@ int main(int argc, char** argv)
 		}
 	}
 
+	if (optind == argc)
+	{
+		tty = strdup(DEFAULT_TTY);
```

Comment:
I've tried to avoid any dynamic memory allocation in this code. Since tty is only used in this scope I think it should be reasonable to avoid the use of the strdup() here.